### ISSUE TITLE
Swagger installation issues

### ISSUE THREAD

**Issue by aroslov** [go-swagger/go-swagger#554]
## Problem statement

I've installed go-swagger using brew (first time), and then faced many issues with package renaming, as the generated code had many packages missing. Having gone through 'migration' section, I've changed it using a script, but still was having issues with things like spec.Document and spec.SecurityScheme being moved around. As these files are auto-generated, seemed like too much of an overhead to change them each time.

Decided to uninstall and tried to re-install from sources, but now the 'swagger' alias is gone. How do I add it?

What am I doing wrong in general? Just seems like a simple example in http://goswagger.io/tutorial/todo-list/ doesn't work no matter what I try

```
$ swagger generate server -A TodoList -f ./swagger.yml 
2016/06/03 13:40:36 building a plan for generation
2016/06/03 13:40:36 planning definitions
2016/06/03 13:40:36 planning operations
2016/06/03 13:40:36 grouping operations into packages
2016/06/03 13:40:36 planning meta data and facades
2016/06/03 13:40:36 rendering 2 models
2016/06/03 13:40:36 rendered model template: error
2016/06/03 13:40:36 rendered model template: item
2016/06/03 13:40:36 rendered handler template: todos.AddOne
2016/06/03 13:40:36 generated handler todos.AddOne
2016/06/03 13:40:36 rendered parameters template: todos.AddOneParameters
2016/06/03 13:40:37 generated parameters todos.AddOneParameters
2016/06/03 13:40:37 rendered responses template: todos.AddOneResponses
2016/06/03 13:40:37 generated responses todos.AddOneResponses
2016/06/03 13:40:37 rendered handler template: todos.DestroyOne
2016/06/03 13:40:37 generated handler todos.DestroyOne
2016/06/03 13:40:37 rendered parameters template: todos.DestroyOneParameters
2016/06/03 13:40:37 generated parameters todos.DestroyOneParameters
2016/06/03 13:40:37 rendered responses template: todos.DestroyOneResponses
2016/06/03 13:40:37 generated responses todos.DestroyOneResponses
2016/06/03 13:40:37 rendered handler template: todos.Find
2016/06/03 13:40:37 generated handler todos.Find
2016/06/03 13:40:37 rendered parameters template: todos.FindParameters
2016/06/03 13:40:37 generated parameters todos.FindParameters
2016/06/03 13:40:37 rendered responses template: todos.FindResponses
2016/06/03 13:40:37 generated responses todos.FindResponses
2016/06/03 13:40:37 rendered handler template: todos.UpdateOne
2016/06/03 13:40:37 generated handler todos.UpdateOne
2016/06/03 13:40:37 rendered parameters template: todos.UpdateOneParameters
2016/06/03 13:40:37 generated parameters todos.UpdateOneParameters
2016/06/03 13:40:37 rendered responses template: todos.UpdateOneResponses
2016/06/03 13:40:37 generated responses todos.UpdateOneResponses
2016/06/03 13:40:37 rendered embedded Swagger JSON template: restapi.TodoList
2016/06/03 13:40:37 rendered builder template: operations.TodoList
2016/06/03 13:40:37 rendered server template: restapi.Server
2016/06/03 13:40:37 rendered configure api template: operations.ConfigureTodoList
2016/06/03 13:40:37 rendered doc template: operations.TodoList
2016/06/03 13:40:37 rendered main template: server.TodoList

$ find .
.
./cmd
./cmd/todo-list-server
./cmd/todo-list-server/main.go
./models
./models/error.go
./models/item.go
./restapi
./restapi/configure_todo_list.go
./restapi/doc.go
./restapi/embedded_spec.go
./restapi/operations
./restapi/operations/todo_list_api.go
./restapi/operations/todos
./restapi/operations/todos/add_one.go
./restapi/operations/todos/add_one_parameters.go
./restapi/operations/todos/add_one_responses.go
./restapi/operations/todos/destroy_one.go
./restapi/operations/todos/destroy_one_parameters.go
./restapi/operations/todos/destroy_one_responses.go
./restapi/operations/todos/find.go
./restapi/operations/todos/find_parameters.go
./restapi/operations/todos/find_responses.go
./restapi/operations/todos/update_one.go
./restapi/operations/todos/update_one_parameters.go
./restapi/operations/todos/update_one_responses.go
./restapi/server.go
./swagger.yml

$ go run ./cmd/todo-list-server/main.go
models/error.go:9:2: cannot find package "github.com/go-swagger/go-swagger/errors" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/errors (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/errors (from $GOPATH)
restapi/operations/todos/add_one_responses.go:9:2: cannot find package "github.com/go-swagger/go-swagger/httpkit" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/httpkit (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/httpkit (from $GOPATH)
restapi/operations/todos/add_one.go:9:2: cannot find package "github.com/go-swagger/go-swagger/httpkit/middleware" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/httpkit/middleware (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/httpkit/middleware (from $GOPATH)
restapi/operations/todo_list_api.go:14:2: cannot find package "github.com/go-swagger/go-swagger/httpkit/security" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/httpkit/security (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/httpkit/security (from $GOPATH)
models/error.go:10:2: cannot find package "github.com/go-swagger/go-swagger/httpkit/validate" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/httpkit/validate (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/httpkit/validate (from $GOPATH)
restapi/operations/todo_list_api.go:15:2: cannot find package "github.com/go-swagger/go-swagger/spec" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/spec (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/spec (from $GOPATH)
models/error.go:7:2: cannot find package "github.com/go-swagger/go-swagger/strfmt" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/strfmt (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/strfmt (from $GOPATH)
restapi/operations/todos/find_parameters.go:14:2: cannot find package "github.com/go-swagger/go-swagger/swag" in any of:
        /usr/local/go/src/github.com/go-swagger/go-swagger/swag (from $GOROOT)
        /Users/anton/go/src/github.com/go-swagger/go-swagger/swag (from $GOPATH)
```

After replacing paths, still:

```
$ go run ./cmd/todo-list-server/main.go
restapi/operations/todo_list_api.go:37: undefined: spec.Document
```

**Comment by vburenin** [comment]
You are running very old version of go-swagger, please consider updating it. Most of those packages that you see are moved to go-openapi relatively long time ago.

**Comment by aroslov** [comment]
More than happy to do it - but how? Brew seems to install an old one. The github version doesn't create the alias for the 'swagger' command to run the generator.

**Comment by vburenin** [comment]
Just install swagger from sources: go get -u github.com/go-swagger/go-swagger/cmd/swagger

**Comment by vburenin** [comment]
anyway, you will still need to pull all dependencies.

**Comment by aroslov** [comment]
Yes, I've done that, how do I get swagger in command line? Symlink?

**Comment by vburenin** [comment]
your binary is installed into GOBIN. So, just add appropriate path into your PATH.

**Comment by aroslov** [comment]
Thank a lot, it's all working now!

Perhaps you should kill the brew version - it only confuses things.

**Comment by mariusmagureanu** [comment]
I'm digging dead bodies here - I ran into the same problem as @aroslov irw 
`undefined: spec.Document`

I've installed swagger from sources, pulled dependencies and did the proper path replacements. 
`swagger version` returns **0.5.0**

I've looked a bit into the **spec** package, there's no Document type - hence the error, somewhere I guess I'm way off with code versions!

Anything I may still be overlooking here? Thanks.
